native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBResultParameterProcessor.java

```diff
@@ -132,4 +139,125 @@ public BObject getBalStreamResultIterator() {
+                case TypeTags.STRING_TAG:
+                    return fromString(timestamp.toString());
+                case TypeTags.OBJECT_TYPE_TAG:
+                case TypeTags.RECORD_TYPE_TAG:
+                    if (type.getName().equalsIgnoreCase(io.ballerina.stdlib.time.util.Constants.CIVIL_RECORD)
+                            && timestamp instanceof Timestamp) {
+                        return Utils.createTimestampRecord((Timestamp) timestamp);
+                    } else if (type.getName().equalsIgnoreCase(io.ballerina.stdlib.time.util.Constants.DATE_RECORD)
+                            && timestamp instanceof Timestamp) {
+                        return Utils.createDateRecord(new java.sql.Date(timestamp.getTime()));
+                    } else {
+                        throw new ApplicationError("Unsupported Ballerina type:" +
+                                type.getName() + " for SQL Timestamp data type.");
+                    }
+                case TypeTags.INT_TAG:
+                    return timestamp.getTime();
+                case TypeTags.INTERSECTION_TAG:
+                    return Utils.createTimeStruct(timestamp.getTime());
+            }
+        }
+        return null;
+    }
+
+    private Object processIntervalResult(ResultSet resultSet, int columnIndex, int sqlType, Type ballerinaType,
+                                         String sqlTypeName) throws ApplicationError, SQLException {
+        String intervalString = resultSet.getString(columnIndex);
+        return convertInterval(intervalString, sqlType, ballerinaType, sqlTypeName);
+    }
+
+    private Object convertInterval(String interval, int sqlType, Type ballerinaType, String sqlTypeName)
+            throws ApplicationError {
+        if (interval != null) {
+            switch (ballerinaType.getTag()) {
+                case TypeTags.STRING_TAG:
+                    return  fromString(interval);
+                case TypeTags.OBJECT_TYPE_TAG:
+                case TypeTags.RECORD_TYPE_TAG:
+                    try {
+                        if (sqlType == -104) {
+                            //format: DD HH:Min:SS.XXX
+                            if (ballerinaType.getName().
+                                    equalsIgnoreCase(io.ballerina.stdlib.time.util.Constants.TIME_OF_DAY_RECORD)) {
+                                Time time = Time.valueOf(interval.split("\\s+")[1].split("\\.")[0]);
+                                return Utils.createTimeRecord(time);
+                            } else if (ballerinaType.getName().
+                                    equalsIgnoreCase(Constants.Types.INTERVAL_DAY_TO_SECOND_RECORD)) {
+                                String[] splitOnSpaces = interval.split("\\s+");
+                                String days = splitOnSpaces[0];
+                                String[] splitOnDots = splitOnSpaces[1].split("\\.");
+                                String xxx = splitOnDots[1];
```

Comment:
Give meaningful names for the variable names

Reply:
added in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/13/commits/950f2228fe9fde68f2865688c03597cabbea3792